be1-go/hub/organizer.go

```diff
@@ -484,3 +491,186 @@ func (c *laoChannel) processRollCallObject(data message.Data) error {
 
 	return nil
 }
+func (c *laoChannel) processElectionObject(msg message.Message) error {
+	action := message.ElectionAction(msg.Data.GetAction())
+
+	if action != message.ElectionSetupAction {
+		return &message.Error{
+			Code:        -1,
+			Description: fmt.Sprintf("invalid action: %s", action),
+		}
+	}
+	err := c.createElection(msg)
+	if err != nil {
+		return xerrors.Errorf("failed to setup the election %v", err)
+	}
+
+	return nil
+}
+
+func (c *laoChannel) createElection(msg message.Message) error {
+	o := c.hub
+
+	o.Lock()
+	defer o.Unlock()
+
+	// Check the data
+	data, ok := msg.Data.(*message.ElectionSetupData)
+	if !ok {
+		return &message.Error{
+			Code:        -4,
+			Description: "failed to cast data to SetupElectionData",
+		}
+	}
+
+	// Check if the Lao ID of the message corresponds to the channel ID
+	encodedLaoID := base64.StdEncoding.EncodeToString(data.LaoID)
+	channelID := c.channelID[6:]
+	if channelID != encodedLaoID {
+		return &message.Error{
+			Code:        -4,
+			Description: fmt.Sprintf("Lao ID of the message (Lao: %s) is different from the channelID (channel: %s)", encodedLaoID, channelID),
+		}
+	}
+
+	// Compute the new election channel id
+	encodedElectionID := base64.StdEncoding.EncodeToString(data.ID)
+	encodedID := encodedLaoID + "/" + encodedElectionID
+
+	// Create the new election channel
+	electionCh := electionChannel{
+		createBaseChannel(o, "/root/"+encodedID),
+		//TODO : check if this is what was meant by method ot is it rather pluralityy and stuff??,
+		//Also can it happen that there is a valid setup without questions?
+		data.StartTime,
+		data.EndTime,
+		false,
+		getAllQuestionsForElectionChannel(data.Questions,data),
+	}
+
+	// Add the SetupElection message to the new election channel
+	messageID := base64.StdEncoding.EncodeToString(msg.MessageID)
+	electionCh.inbox[messageID] = msg
+
+	// Add the new election channel to the organizerHub
+	o.channelByID[encodedID] = &electionCh
+
+	return nil
+}
+
+func getAllQuestionsForElectionChannel(questions []message.Question,data *message.ElectionSetupData)map[string]question{
+	qs := make(map[string]question)
+	for _,q := range questions{
+		qs[ base64.StdEncoding.EncodeToString(q.ID)] = question{
+			q.ID,
+			q.BallotOptions,
+			make(map[string]validVote),
+			data.Questions[0].VotingMethod,
```

Comment:
Yes you are completely right I think this was before the voting method was in the question structure of the election channel but I didn't change it since !